opt_adapt/opt.py

```diff
@@ -116,15 +116,15 @@ def _BFGS(forward_run, m, params, u, u_, dJ_, B, Rspace=False):
                 s -= u_
                 y = dJ.copy(deepcopy=True)
                 y -= dJ_
-                lr = s/y
+                lr = s.dat.data[0]/y.dat.data[0]
```

Comment:
In the case of functions from R-space, you can actually put `lr = float(s) / float(y)`.